client/app/components/SearchBar/SearchBar.js

```diff
@@ -13,16 +15,139 @@ const SEARCH_MODES = [
   SEARCH_MODE_KEYWORD_AND_LOCATION,
 ];
 
+const coordinates = (place) => {
+  if (place && place.geometry) {
+    return place.geometry.location.toUrlValue();
+  }
+  return null;
+};
+
+const getDetails = (placeId) => new Promise((resolve, reject) => {
+  const serviceStatus = window.google.maps.places.PlacesServiceStatus;
+  const div = document.createElement('div');
+  const service = new window.google.maps.places.PlacesService(div);
+
+  service.getDetails({ placeId }, (place, status) => {
+    if (status !== serviceStatus.OK) {
+      reject(new Error(`Could not get details for place id "${placeId}"`));
+    } else {
+      resolve(place);
+    }
+  });
+});
+
+const getPrediction = (location) => new Promise((resolve, reject) => {
+  const serviceStatus = window.google.maps.places.PlacesServiceStatus;
+  const service = new window.google.maps.places.AutocompleteService();
+
+  service.getPlacePredictions({ input: location }, (predictions, status) => {
+    if (status !== serviceStatus.OK) {
+      reject(new Error(`Prediction service status not OK: ${status}`));
+    } else if (predictions.length === 0) {
+      reject(new Error(`No predictions found for location "${location}"`));
+    } else {
+      resolve(getDetails(predictions[0].place_id));
+    }
+  });
+});
+
 class SearchBar extends Component {
   constructor(props) {
     super(props);
 
-    // The values of the inputs are synced to the component state
-    // until the form is submitted.
-    this.state = {
-      keyword: '',
-      location: '',
-    };
+    // Bind `this` within methods
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+    this.state = { selectedPlace: null };
+  }
+  componentDidMount() {
+    document.body.classList.add(css.topLevelBody);
+    if (!this.locationInput) {
+      return;
+    }
+    const bounds = { north: -90, east: -180, south: 90, west: 180 };
+    const autocomplete = new window.google.maps.places.Autocomplete(this.locationInput, { bounds });
```

Comment:
I just verified that the `componentDidMount` isn't called when rendering in the server side, so no changes needed here.